lib/runner/nested-request.js

```diff
@@ -39,8 +39,23 @@ function runNestedRequest ({ executionId, isExecutionSkipped, vaultSecrets, item
                 ' calls have been reached for this request.'));
         }
 
+        let rootRequestEventsList = self.state.nestedRequest.rootItem.events?.all?.() || [],
+            requestMetadataFromItem = null,
+            additionalInfo = { rootRequestId: self.state.nestedRequest.rootItem.id };
+
+        for (const event of rootRequestEventsList) {
+            if (event.script && event.script.requests && event.script.requests[requestToRunId]) {
+                requestMetadataFromItem = event.script.requests[requestToRunId];
+                break;
+            }
+        }
+
+        if (requestMetadataFromItem) {
+            additionalInfo.requestMetadataFromItem = requestMetadataFromItem;
+        }
+
         // Should fetch the request from the consumer of postman-runtime & resolve scripts and variables
-        requestResolver(requestToRunId, function (err, collection) {
+        requestResolver(requestToRunId, additionalInfo, function (err, collection) {
```

Comment:
```js
context = { rootItemId, referencedRequestMeta: { workspace, collection, path } }
```
- rootItemId since runtime understands item and not request, so keeping it consistent with that
- if we are using `nestedRequest`, we can call it `nestedRequestMeta` as well